src/patternfly/components/NotificationBadge/notification-badge.scss

```diff
@@ -39,17 +56,38 @@
     transform: translate(var(--pf-c-notification-badge--after--TranslateX), var(--pf-c-notification-badge--after--TranslateY));
   }
 
+  .pf-c-button & {
+    position: static;
+  }
+
   & > i {
     width: var(--pf-c-notification-badge__i--Width);
     height: var(--pf-c-notification-badge__i--Height);
   }
 
-  &.pf-m-unread::after {
+  & > * {
+    position: relative;
+  }
+
+  &.pf-m-unread {
     --pf-c-notification-badge--after--BackgroundColor: var(--pf-c-notification-badge--m-unread--after--BackgroundColor);
+    --pf-c-notification-badge--after--BorderWidth: var(--pf-c-notification-badge--m-unread--after--BorderWidth);
+    --pf-c-notification-badge--after--BorderColor: var(--pf-c-notification-badge--m-unread--after--BorderColor);
   }
 
-  &.pf-m-read::after {
+  &.pf-m-read {
     --pf-c-notification-badge--after--BackgroundColor: var(--pf-c-notification-badge--m-read--after--BackgroundColor);
     --pf-c-notification-badge--after--BorderColor: var(--pf-c-notification-badge--m-read--after--BorderColor);
   }
+
+  &.pf-m-attention {
+    --pf-c-notification-badge--after--BackgroundColor: var(--pf-c-notification-badge--m-attention--after--BackgroundColor);
```

Comment:
should we add this and any other background color modifications to `::before`?

Reply:
Yeah, GH is collapsing the code https://github.com/patternfly/patternfly/blob/5c62a351df736a71755d51e9204135ce0d264781/src/patternfly/components/NotificationBadge/notification-badge.scss#L52